### PR TITLE
fix(rome_js_formatter): make empty line after hashbang optional

### LINKED ISSUE #3497: 🐛 Rome enforces an empty newline after a hashbang, prettier doesn't
### Environment information

```bash
OS: win11, 64bit
installation method: `npm i -D rome@0.10.1-next.e8dd9b4`
```

### What happened?

1. Open this [playground](https://play.rome.tools/?code=%23%21%2Fusr%2Fbin%2Fenv+node%0Aimport+fs+from+%27node%3Afs%27%3B&lineWidth=80&indentStyle=tab&quoteStyle=double&quoteProperties=as-needed&trailingComma=all&indentWidth=2&sourceType=module&enabledNurseryRules=true&typescript=true&jsx=false#IwAhAC8AdQBzAHIALwBiAGkAbgAvAGUAbgB2ACAAbgBvAGQAZQAKAGkAbQBwAG8AcgB0ACAAZgBzACAAZgByAG8AbQAgACcAbgBvAGQAZQA6AGYAcwAnADsA)
2. Notice that Rome always enforces an empty newline after a [hashbang](https://github.com/tc39/proposal-hashbang), while prettier allows 0 or 1 newline

### Expected result

Prettier compatibility.

TBH this is very minor, and I don't have a strong opinion about this, but maybe someone else does.

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

### PR DESCRIPTION
**Pull request by unvalley**
## Summary
Closes #3497

## Test Plan
I updated snapshot results by `cargo insta review`.

### CODE REVIEW

File: crates/rome_js_formatter/src/utils/mod.rs [hunk 1]
```diff
@@ -124,7 +124,11 @@ impl<'a> FormatInterpreterToken<'a> {
 impl Format<JsFormatContext> for FormatInterpreterToken<'_> {
     fn fmt(&self, f: &mut JsFormatter) -> FormatResult<()> {
         if let Some(interpreter) = self.token {
-            write!(f, [interpreter.format(), empty_line()])
+            if interpreter.kind() == JsSyntaxKind::JS_SHEBANG {
+                write!(f, [interpreter.format(), hard_line_break()])
+            } else {
+                write!(f, [interpreter.format(), empty_line()])
+            }
```
Comment (unvalley) [inline crates/rome_js_formatter/src/utils/mod.rs:131]: Honestly speaking, I don't have confidence that the changes here are reasonable. 
This file is under the `utils/`, Is `JS_SHEBANG` allowed to appear here?
Please let me know, if this change is not correct direction.

File: crates/rome_js_formatter/src/utils/mod.rs [hunk 2]
```diff
@@ -124,7 +125,11 @@ impl<'a> FormatInterpreterToken<'a> {
...
```
Comment (MichaReiser) [inline crates/rome_js_formatter/src/utils/mod.rs:132]: I don't think checking for the `JS_SHEBANG` kind is necessary. `FormatInterpreterToken` is used when formatting a module or script to format the interpreter/shebang token (that's some inconsistent naming from my side). 

The decision whether an empty line or a single hard line break must be inserted depends if there's an empty line "after" the interpreter/shebang token as you can see with the following examples:

```javascript
#!/usr/bin/env


console.log("Test");

// Should become
#!/usr/bin/env

console.log("Test");
```


```javascript
#!/usr/bin/env
console.log("Test");

// Should become
#!/usr/bin/env
console.log("Test");
```

You can test if there's an empty line by retrieving the next token (there should be a method on the interpreter token) and then use `get_lines_before_token` to count how many newlines separate the current and next token (new lines are always associated with the next token)

File: crates/rome_js_formatter/src/utils/mod.rs [hunk 3]
```diff
@@ -124,7 +124,16 @@ impl<'a> FormatInterpreterToken<'a> {
 impl Format<JsFormatContext> for FormatInterpreterToken<'_> {
     fn fmt(&self, f: &mut JsFormatter) -> FormatResult<()> {
         if let Some(interpreter) = self.token {
-            write!(f, [interpreter.format(), empty_line()])
+            if let Some(token) = interpreter.next_token() {
+                // the interpreter line + empty line = 2
+                if get_lines_before_token(&token) >= 2 {
+                    write!(f, [interpreter.format(), empty_line()])
```
Comment (unvalley) [inline crates/rome_js_formatter/src/utils/mod.rs:130]: In the following case, I considered the presence of two or more lines before the token to be a requirement.
```
#!/usr/bin/env


console.log("Test");

// Should become
#!/usr/bin/env

console.log("Test");
```

File: crates/rome_js_formatter/src/utils/mod.rs (hunk 2 above)
Comment (unvalley) [reply crates/rome_js_formatter/src/utils/mod.rs]: @MichaReiser 
Thank you for helpful explanation. I edited code to use `interpreter.next_token` and `get_lines_before_token`.
And I added the test for first example.

File: crates/rome_js_formatter/src/utils/mod.rs [hunk 4]
```diff
@@ -124,7 +124,16 @@ impl<'a> FormatInterpreterToken<'a> {
...
+                } else {
+                    write!(f, [interpreter.format(), hard_line_break()])
+                }
+            } else {
+                Ok(())
+            }
         } else {
```
[the remark this answers is not in the document]
Comment (unvalley) [reply crates/rome_js_formatter/src/utils/mod.rs]: Thanks for reviewing! I've fixed the code at [463bd03](https://github.com/rome/tools/pull/3552/commits/463bd03e3e592ca86628822a64bae4ae99b79c9a).

**Review by MichaReiser — APPROVED** [review]
Nice! Thanks for fixing this issue

**Comment by unvalley** [comment]
I've learned a lot!

**Pull request merged**